udata_front/theme/gouvfr/assets/js/components/ProducerSelector/ProducerSelector.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <div v-if="isAdmin">
+    <MultiSelect
+      :required="true"
+      :minimumCharacterBeforeSuggest="2"
+      :placeholder="$t('Check the identity with which you want to publish')"
+      :searchPlaceholder="$t('Select an organization')"
+      suggestUrl="/organizations/suggest/"
+      :initialOptions="organizations"
+      :values="userOrganization"
+      @change="updateOrganization"
+    />
+  </div>
+  <div v-else-if="hasOrganizations">
+    <MultiSelect
+      :required="true"
+      :minimumCharacterBeforeSuggest="2"
+      :placeholder="$t('Check the identity with which you want to publish')"
+      :searchPlaceholder="$t('Select an organization')"
+      :initialOptions="organizations"
+      :values="userOrganization"
+      @change="updateOrganization"
+    />
+  </div>
+  <div v-else>
+    <div class="fr-col bg-contrast-grey text-align-center fr-p-2v">
+      <p class="fr-text--md fr-text--bold fr-mb-n3v">{{ t('You belong to no organization') }}</p>
+      <p class="fr-text--sm fr-text--bold fr-mb-n4v">{{ t('You publish in your own name') }}</p>
+      <p class="fr-text--sm">{{ t('We advise you to publish under an organization if it\'s a professional activity') }}</p>
+      <div class="fr-grid-row fr-grid-row--middle fr-pb-3v">
+        <div class="fr-col-6">
+          <a class="fr-btn fr-btn--secondary fr-btn--secondary-grey-500" :href="organizationsUrl">
+            {{ $t("Join an organization") }}
+          </a>
+        </div>
+        <div class="fr-col-6">
+          <a class="fr-btn" :href="createOrganizationUrl">
+            {{ $t("Create an organization") }}
+          </a>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { ref, computed, toValue, onMounted } from 'vue';
+import MultiSelect from '../../components/MultiSelect/MultiSelect.vue';
+import type { User, Organization } from '@etalab/data.gouv.fr-components';
+import { organization_url } from '../../config';
+import { useI18n } from 'vue-i18n';
+import useUserAvatar from "../../composables/useUserAvatar";
+
+const { t } = useI18n();
+
+const props = defineProps<{user: User}>();
+
+const organizationsUrl = organization_url;
+const createOrganizationUrl = `${organization_url}publishing-form/`;
+
+const userOrganization = ref<Organization>();
+const organizations = ref<Array<Organization>>([]);
```

Comment:
Did you try to type it with `Organization | User` ? Does it work ? This would allow us to use `user` directly instead of mapping it to an organization